src/wanderer.ts

```diff
@@ -46,165 +52,200 @@ export function digitizedMonstersRemaining(): number {
+      : 0) +
+    (testZoneForBackups(zone)
+      ? have($item`backup camera`)
+        ? clamp(11 - get("_backUpUses"), 0, 11)
+        : 0
+      : 0)
   );
+}
 
-  acceptBestGuzzlrQuest();
-
-  const currentGuzzlrZone = Guzzlr.getLocation() || $location`none`;
-  if (
-    !testZoneAndUsePotionToAccess() ||
-    (!testZoneForWanderers(currentGuzzlrZone) &&
-      predictedWanderers > predictedBackups &&
-      predictedBackups < turnsLeftOnThisQuest) ||
-    (!testZoneForBackups(currentGuzzlrZone) && predictedBackups >= predictedWanderers)
-  ) {
-    Guzzlr.abandon();
-  }
-  acceptBestGuzzlrQuest();
-
-  const guzzlZone = Guzzlr.getLocation();
-  if (!testZoneAndUsePotionToAccess()) return defaultLocation;
-  if (
-    !guzzlZone ||
-    (type === draggableFight.WANDERER && !testZoneForWanderers(guzzlZone)) ||
-    (type === draggableFight.BACKUP && !testZoneForBackups(guzzlZone))
-  ) {
-    return defaultLocation;
+function freeCrafts() {
+  return (
+    (have($skill`Rapid Prototyping`) ? 5 - get("_rapidPrototypingUsed") : 0) +
+    (have($skill`Expert Corner-Cutter`) ? 5 - get("_expertCornerCutterUsed") : 0)
+  );
+}
+
+function prepareGuzzlr(): boolean {
+  const hasFreeCrafts = freeCrafts() > 0;
+  if (!Guzzlr.isQuestActive()) {
+    // const platBooze = Guzzlr.getCheapestPlatinumCocktail(hasFreeCrafts);
+    const platBooze = Guzzlr.getCheapestPlatinumCocktail();
+    if (
+      Guzzlr.canPlatinum() &&
+      (have(platBooze) || freeCrafts() > 0 || buy(1, platBooze, WANDERER_PRICE_THRESHOLD) > 0) &&
+      (!Guzzlr.haveFullPlatinumBonus() ||
+        (Guzzlr.haveFullBronzeBonus() && Guzzlr.haveFullGoldBonus()))
+    ) {
+      Guzzlr.acceptPlatinum();
+    } else if (Guzzlr.canGold() && (!Guzzlr.haveFullGoldBonus() || Guzzlr.haveFullBronzeBonus())) {
+      Guzzlr.acceptGold();
+    } else {
+      Guzzlr.acceptBronze();
+    }
   }
 
-  const choicesToSet = unsupportedChoices.get(guzzlZone);
-  if (choicesToSet) propertyManager.setChoices(choicesToSet);
+  const location = Guzzlr.getLocation()!;
+  const guzzlrBooze =
+    Guzzlr.getTier() === "platinum"
+      ? Guzzlr.getCheapestPlatinumCocktail() // Guzzlr.getCheapestPlatinumCocktail(hasFreeCrafts)
+      : Guzzlr.getBooze();
 
-  if (Guzzlr.getTier() === "platinum") {
-    zonePotions.forEach((place) => {
-      if (guzzlZone.zone === place.zone && !have(place.effect)) {
-        if (!have(place.potion)) {
-          buy(1, place.potion, 10000);
-        }
-        use(1, place.potion);
-      }
-    });
-    if (!Guzzlr.havePlatinumBooze()) {
-      print("It's time to get buttery", "purple");
-      cliExecute("make buttery boy");
-    }
-  } else {
-    const guzzlrBooze = Guzzlr.getBooze();
-    if (!guzzlrBooze) {
-      return defaultLocation;
-    } else if (!have(guzzlrBooze)) {
-      print("just picking up some booze before we roll", "blue");
+  // error state - accepted a quest, but mafia says the quest location or booze is null
```

Comment:
Consider doing `refresh quests` if we end up in this state. Ultimately we'll need _some_ sort of catch, just for typescript's sake, but we might as well